django/db/models/options.py

```diff
@@ -354,6 +336,21 @@ def swapped(self):
                     return swapped_for
         return None
 
+    @property
```

Comment:
Well, I think the staleness issue could be easily addressed by having `Manager.contribute_to_class` invalidate the cache.

But you're right, this is probably only worth it if we try my other suggestion of trying to reuse this at runtime instead of re-copying on every access. And _that_ is probably only worth it if we can show that the copying-at-runtime is a noticeable perf regression.